python/Ganga/Core/GangaThread/WorkerThreads/WorkerThreadPool.py

```diff
@@ -226,10 +226,11 @@ def _stop_worker_threads(self, shutdown=False):
         self._shutdown = shutdown
         for w in self.__worker_threads:
             w.stop()
+            w.join()
             # FIXME NEED TO CALL AN OPTIONAL CLEANUP FUCNTION HERE IF THREAD IS STOPPED
             # w.unregister()
             #del w
-        #del self.__worker_threads[:]
+        del self.__worker_threads[:]
```

Comment:
There should be no need to `del` all the entries here as you're rebinding the list on the next line so any remaining reference to them should be gone.